ionc/ion_reader_binary.c

```diff
@@ -368,8 +368,8 @@ iERR _ion_reader_binary_step_out(ION_READER *preader)
         ASSERT(preader->_eof ? (curr_pos == next_start) : (curr_pos <= next_start));  
         to_skip = next_start - curr_pos;
         while (to_skip > 0) {
-            if (to_skip > SIZE_MAX) {
-                one_skip = SIZE_MAX;
+            if (to_skip > MAX_SIZE) {
+                one_skip = MAX_SIZE;
```

Comment:
to_skip is int64_t, and one_skip is SIZE which is int32_t. SIZE_MAX is a built-in macro in C which is of type int64_t, whereas MAX_SIZE is an Ion macro of type int32_t set to INT32_MAX. Since we were unintentionally downcasting and it's pretty obvious that our SIZE and MAX_SIZE macros are intended to commingle this was an easy fix